flow/envs/multiagent/i210.py

```diff
@@ -156,53 +159,67 @@ def compute_reward(self, rl_actions, **kwargs):
                 if self.k.vehicle.get_speed(rl_id) >= 0:
                     speeds.append(self.k.vehicle.get_speed(rl_id))
                 if len(speeds) > 0:
-                    # rescale so the q function can estimate it quickly
+                    # rescale so the critic can estimate it quickly
                     rewards[rl_id] = np.mean([(des_speed - np.abs(speed - des_speed))**2
                                               for speed in speeds]) / (des_speed**2)
         else:
-            for rl_id in self.k.vehicle.get_rl_ids():
-                if self.env_params.evaluate:
-                    # reward is speed of vehicle if we are in evaluation mode
-                    reward = self.k.vehicle.get_speed(rl_id)
-                elif kwargs['fail']:
-                    # reward is 0 if a collision occurred
-                    reward = 0
-                else:
-                    # reward high system-level velocities
-                    cost1 = average_velocity(self, fail=kwargs['fail'])
-
-                    # penalize small time headways
-                    cost2 = 0
-                    t_min = 1  # smallest acceptable time headway
-
-                    lead_id = self.k.vehicle.get_leader(rl_id)
-                    if lead_id not in ["", None] \
-                            and self.k.vehicle.get_speed(rl_id) > 0:
-                        t_headway = max(
-                            self.k.vehicle.get_headway(rl_id) /
-                            self.k.vehicle.get_speed(rl_id), 0)
-                        cost2 += min((t_headway - t_min) / t_min, 0)
-
-                    # weights for cost1, cost2, and cost3, respectively
-                    eta1, eta2 = 1.00, 0.10
-
-                    reward = max(eta1 * cost1 + eta2 * cost2, 0)
-
-                rewards[rl_id] = reward
+            speeds = self.k.vehicle.get_speed(self.k.vehicle.get_ids())
+            des_speed = self.env_params.additional_params["target_velocity"]
+            # rescale so the critic can estimate it quickly
+            reward = np.nan_to_num(np.mean([(des_speed - np.abs(speed - des_speed))**2
+                                              for speed in speeds]) / (des_speed**2))
+            rewards = {rl_id: reward for rl_id in self.k.vehicle.get_rl_ids()}
         return rewards
 
     def additional_command(self):
         """See parent class.
 
-        Define which vehicles are observed for visualization purposes.
+        Define which vehicles are observed for visualization purposes. Additionally, optionally reroute vehicles
+        back once they have exited.
         """
+        super().additional_command()
```

Comment:
Why is this necessary? If I"m not mistaken, super's additional command is defined to be an abstract method (the body is just 'pass')

Reply:
It's just good practice to call the super() method, no?